### ISSUE TITLE
Strapi Graphql returns an error after register new user

### ISSUE THREAD

**Issue by OlegMarchenko** [strapi/strapi#7766]
**Steps to reproduce the behavior**

**Expected behavior**

**Screenshots**

**Code snippets**

```
mutation registerUser (
        $username: String!
        $email: String!
        $password: String!
    ) {
        register (
            input: {
                username: $username
                email: $email
                password: $password
            }
        ) {
            jwt
            user {
                username 
                email
                password
            }
        }
    }
```

**System**

- Node.js version: 12.18.2
- Strapi version: 3.1.1
- Database: Mongo
- Operating system: Windows

**Additional context**
Graphql returned an error, but new user appears in strapi backend users
https://www.loom.com/share/fe4c25a5992c4a6498e071c45a746b00

**Comment by derrickmehaffy** [comment]
As mentioned on slack, please try upgrading to the latest release `v3.1.4`

**Comment by OlegMarchenko** [comment]
> As mentioned on slack, please try upgrading to the latest release `v3.1.4`

done, but still nothing

**Comment by OlegMarchenko** [comment]
okey

**Comment by richardgrey** [comment]
I was not able to reproduce this bug on v3.1.4. Was trying also to play with different permissions for the user from which I send the requests, but the system responds correctly with forbidden status. Anyone else?

**Comment by lauriejim** [comment]
Hello! Did you customize your User model?
And second thing, did you activate any options in the Roles & Permissions section.
Espcialy un the advanced section?

**Comment by derrickmehaffy** [comment]
@lauriejim I was only able to reproduce it on Mongo

**Comment by onsombal** [comment]
I'm using Strapi 3.1.4 and encountering the same issue — graqphql "register" user adds a new user, but never returns the JWT token and errors out. I tested in the playground and in code; both having the same issue with returning the JWT token.

Playground code:
```
mutation register($input: UsersPermissionsRegisterInput!) {
	register(input: $input) {
    user {
      username
      email
    }
    jwt
  }
}
```
Playground Variables:
```
{
  "input": {
    "username": "mac",
    "email": "mac01@gmail.com",
    "password": "123456"
  }
```
If you review the Playground Error that's returned below, you will notice it is complaining about "email is already taken", which is not true. It seems like, the "register" user mutation is in a loop.
```
{
  "errors": [
    {
      "message": "Bad Request",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "register"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "code": 400,
          "data": {
            "statusCode": 400,
            "error": "Bad Request",
            "message": [
              {
                "messages": [
                  {
                    "id": "Auth.form.error.email.taken",
                    "message": "Email is already taken."
                  }
                ]
              }
            ],
            "data": [
              {
                "messages": [
                  {
                    "id": "Auth.form.error.email.taken",
                    "message": "Email is already taken."
                  }
                ]
              }
            ]
          },
          "stacktrace": [
            "Error: Bad Request",
            "    at checkBadRequest (/Users/jimbiggs/Local/GraphQL_tutorials/max/react-apollo-strapi-signup/backend/node_modules/strapi-plugin-users-permissions/config/schema.graphql.js:11:23)",
            "    at resolver (/Users/jimbiggs/Local/GraphQL_tutorials/max/react-apollo-strapi-signup/backend/node_modules/strapi-plugin-users-permissions/config/schema.graphql.js:196:11)"
          ]
        }
      }
    }
  ],
  "data": null
}

```

**Comment by derrickmehaffy** [comment]
Testing on Strapi v3.2.3

MySQL/SQLite/MariaDB passed just fine:

![image](https://user-images.githubusercontent.com/8593673/95895610-c8544a00-0d3f-11eb-9155-69d6be9b8f9b.png)

Mongo also worked:

![image](https://user-images.githubusercontent.com/8593673/95895977-647e5100-0d40-11eb-9caf-5e570643e083.png)


Going to mark this issue as closed for now unless more information can provided on the latest version (3.2.x+) and the error can be reproduced.

**Comment by onsombal** [comment]
@derrickmehaffy,

I upgraded to Strapi-3.2.3, but still no joy in resolving the error. Below are screenshots. Please note that once I hit the "Play Button" on the playground, the user is registered immediately as viewed in the Playground Users Query, the Strapi UI for Users Collection, and on MongoDB Atlas.

However, the playground "register" view continues to spin for a long while until it finally returns the error I encountered -- that the "**Email is already taken**".

Playground Register view after hitting "Play" to "Register **Sam**":
![image](https://user-images.githubusercontent.com/1588266/95923483-97960400-0d83-11eb-87b5-fb3aef004b85.png)

Playground Users Query View after hitting "Play" to "Register **Sam**". Sam is listed:
![image](https://user-images.githubusercontent.com/1588266/95922804-1722d380-0d82-11eb-8d07-8ae55c870565.png)

Strapi UI for Users view after hitting "Play" to "Register **Sam**". Sam is listed:
![image](https://user-images.githubusercontent.com/1588266/95922857-3a4d8300-0d82-11eb-988f-5690795b2778.png)

Playground Register view after finally timing out, and reporting error — "Email is already taken.":
![image](https://user-images.githubusercontent.com/1588266/95922896-4fc2ad00-0d82-11eb-8a46-9811ca3a655f.png)

MongoDB Atlas view after hitting "Play" to "Register **Sam**". Sam is listed:
![image](https://user-images.githubusercontent.com/1588266/95923141-de372e80-0d82-11eb-99de-9c49281ad4ea.png)

My package.json:
![image](https://user-images.githubusercontent.com/1588266/95923638-efcd0600-0d83-11eb-839c-e5b1996d5957.png)

**Comment by derrickmehaffy** [comment]
We need more information to reproduce, I spun up 4 different projects covering every database and couldn't reproduce.

**Comment by onsombal** [comment]
Not sure what more information I can provide. Did you test using MongoDB Atlas?

**Comment by derrickmehaffy** [comment]
I did test with MongoDB Atlas and couldn't reproduce, are you able to reproduce on a fresh project?

**Comment by jimmy121192** [comment]
I have the same problem. The register isn't returned any data, but the new user is shown up in the user list. I'm using Apollo Client. However,  because of this issue, the update() method didn't trigger after I ran the register mutation.

Would like to hear any solution to solve this issue.

**Comment by onsombal** [comment]
> @onsombal I have the same problem. The register isn't returned any data, but the new user is shown up in the user list. I'm using Apollo Client. However, because of this issue, the update() method didn't trigger after I ran the register mutation.
> 
> Would like to hear any solution to solve this issue.
> Tested on Strapi: v3.4.3

Hi Jimmy, I never solved the issue. I ended up using the REST login for Registration. Everything else in my app uses @apollo/client v3

The REST approach for logging in, resetting passords, etc is simple and quick, so I'm fine with this approach.

**Comment by jimmy121192** [comment]
@onsombal Thanks for your reply. I was able to find out why this issue happened. **It was because the Email Confirmation was ON. So by turning this feature OFF, I was able to receive a response from the server.** 
I have done some tests, and I realized that the Email Confirmation only worked when the Strapi project was deployed online ( which is understandable). And the Server did send a response after a new user was registered.
